apps/desktop/src/components/v3/IntegrateUpstreamModal.svelte

```diff
@@ -122,27 +123,19 @@
 				.then((result) => {
 					targetCommitOid = result;
 				});
+		} else {
+			// If there is no divergence we should set this to undefined.
+			targetCommitOid = undefined;
 		}
 	});
 
-	// Fetch the reviews for the applied branches
-	$effect(() => {
-		if (!modal?.imports.open) return;
-		if (
-			filteredReviews === undefined &&
-			appliedBranches !== undefined &&
-			forgeListingService !== undefined
-		) {
-			if (appliedBranches.length === 0) {
-				filteredReviews = [];
-				return;
-			}
-
-			forgeListingService.fetchByBranch(projectId, appliedBranches).then((reviews) => {
-				filteredReviews = reviews.filter((r) => r.sourceBranch !== undefined);
-			});
-		}
-	});
+	async function setFilteredBranches(appliedBranches: string[]) {
+		if (!forgeListingService) return;
+
+		const reviews = await forgeListingService.fetchByBranch(projectId, appliedBranches);
+		// Find the reviews that have a "mergedAt" timestamp
+		filteredReviews = reviews.filter((r) => !!r.mergedAt);
+	}
```

Comment:
This is a rewrite of the old effect as a function. We are safe to write this as a function like this because the dependencies of this function were only set during the `open` function call anyway.

One change we did make is in how we filter the reviews. Previously a PR with no `sourceBranch` was considered "merged", but that only works if the GH repository is configured to automatically delete branches. This change makes use of the `mergedAt` property, which is at timestamp of when the PR was merged, and will only ever be set if the PR has in fact been merged.